src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/ConstructorOnTypeBuilderInstantiation.cs

```diff
@@ -58,7 +58,7 @@ public override object Invoke(object? obj, BindingFlags invokeAttr, Binder? bind
         public override CallingConventions CallingConvention => _ctor.CallingConvention;
         public override Type[] GetGenericArguments() { return _ctor.GetGenericArguments(); }
         public override bool IsGenericMethodDefinition => false;
-        public override bool ContainsGenericParameters => false;
+        public override bool ContainsGenericParameters => _ctor.ContainsGenericParameters;
```

Comment:
```suggestion
        public override bool ContainsGenericParameters => _type.ContainsGenericParameters;
```
I think this should match the declaring type. Similar to https://github.com/dotnet/runtime/blob/027a43b92bbdf54efa313d2fdaa6e57e7e088fcc/src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.CoreCLR.cs#L234